pkg/operator/ceph/object/zone/controller.go

```diff
@@ -219,27 +228,39 @@ func (r *ReconcileObjectZone) createCephZone(zone *cephv1.CephObjectZone, realmN
 		}
 	}
 
-	// check if master zone group does not exist yet for period
-	masterZone, err := decodeMasterZone(output)
+	// check if master zone does not exist yet for period
+	zoneGroupJson, err := object.DecodeZoneGroupConfig(output)
 	if err != nil {
-		return reconcile.Result{}, errors.Wrapf(err, "failed to parse `radosgw-admin zonegroup get` output")
+		return reconcile.Result{}, errors.Wrap(err, "failed to parse `radosgw-admin zonegroup get` output")
 	}
 
-	masterArg := ""
-	// master zone does not exist yet for zone group
-	if masterZone == "" {
-		masterArg = "--master"
+	zoneIsMaster := false
+	if zoneGroupJson.MasterZoneID == "" {
+		zoneIsMaster = true
 	}
 
 	// create zone
-	_, err = object.RunAdminCommandNoRealm(objContext, "zone", "get", realmArg, zoneGroupArg, zoneArg)
+	_, err = object.RunAdminCommandNoMultisite(objContext, "zone", "get", realmArg, zoneGroupArg, zoneArg)
 	if err != nil {
 		if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
 			logger.Debugf("ceph zone %q not found, running `radosgw-admin zone create`", zone.Name)
-			_, err := object.RunAdminCommandNoRealm(objContext, "zone", "create", realmArg, zoneGroupArg, zoneArg, masterArg)
+			accessKeyArg, secretKeyArg, err := object.GetRealmKeyArgs(r.context, realmName, zone.Namespace)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "failed to get keys for realm")
+			}
+
+			if zoneIsMaster {
+				// master zone does not exist yet for zone group
+				masterArg := "--master"
+				_, err = object.RunAdminCommandNoMultisite(objContext, "zone", "create", realmArg, zoneGroupArg, zoneArg, accessKeyArg, secretKeyArg, masterArg)
```

Comment:
Instead of running a separate command, create an args slice and pass it to a single command. 
```
args := []string{"zone", "create", <fill in others> }
if zoneIsMaster{
  args = append(args, "--master")
}
_, err = object.RunAdminCommandNoMultisite(objContext, args...)
```